packages/block-editor/src/components/block-navigation/list.js

```diff
@@ -40,30 +40,20 @@ export default function BlockNavigationList( {
 		/* eslint-disable jsx-a11y/no-redundant-roles */
 		<ul className="block-editor-block-navigation__list" role="list">
 			{ map( omitBy( blocks, isNil ), ( block ) => {
-				const blockType = getBlockType( block.name );
 				const isSelected = block.clientId === selectedBlockClientId;
-
+				const itemProps = {
+					block,
+					isSelected,
+					onClick: () => onItemClick( block.clientId )
+				};
 				return (
 					<li key={ block.clientId }>
-						<div className="block-editor-block-navigation__item">
-							<Button
-								className={ classnames(
-									'block-editor-block-navigation__item-button',
-									{
-										'is-selected': isSelected,
-									}
-								) }
-								onClick={ () => selectBlock( block.clientId ) }
-							>
-								<BlockIcon icon={ blockType.icon } showColors />
-								{ getBlockLabel( blockType, block.attributes ) }
-								{ isSelected && (
-									<VisuallyHidden as="span">
-										{ __( '(selected block)' ) }
-									</VisuallyHidden>
-								) }
-							</Button>
-						</div>
+						<ListItemSlot blockId={ block.clientId }>
```

Comment:
You may be able to avoid the need to pass the `clientId` here by using something like `BlockContext` internally in the `ListItemSlot`, this seems to have the `clientId`:
https://github.com/WordPress/gutenberg/blob/6c26cdf1f97307b478e37d731351c4b1125e902b/packages/block-editor/src/components/block-list/block-wrapper.js#L36-L53

Or there might be another way. I think all the other slots rely on showing only content for the selected block, whereas this use case is a bit different.